src/pose_publisher.cpp

```diff
@@ -1,85 +1,130 @@
+        
+        // std::chrono::seconds timer_period(1);
+        std::chrono::milliseconds timer_period{500};
+
+        // Call timer_callback function every timer_period
+        timer_ = this->create_wall_timer(
+            timer_period, std::bind(&PosePubNode::timer_callback, this));
+    }
+
+    void read_parameters()
+    {
+        // Declare and acquire parameters
+        declare_parameter<std::string>("robot", tf_prefix);
+        declare_parameter<std::string>("map", map_frame);
+        declare_parameter<std::string>("base_link", base_frame);
+        declare_parameter<std::string>("pose", topic_republish);
+
+        get_parameter("robot", tf_prefix);
+        get_parameter("map", map_frame);
+        get_parameter("base_link", base_frame);
+        get_parameter("pose", topic_republish);
     }
 
-    if(tf_ok) {
-      geometry_msgs::PoseWithCovarianceStamped pose_stamped;
-      pose_stamped.header.stamp = ros::Time::now();
-      pose_stamped.header.frame_id = tf_prefix+"/"+map_frame;
 
-      pose_stamped.pose.pose.position.x = transform.getOrigin().getX();
-      pose_stamped.pose.pose.position.y = transform.getOrigin().getY();
-      pose_stamped.pose.pose.position.z = transform.getOrigin().getZ();
+private:
+    void timer_callback()
+    {
+        RCLCPP_INFO(get_logger(),  "In timer_callback");
+
+        std::string fromFrameRel = (tf_prefix + "/" + base_frame).c_str();
+        std::string toFrameRel = (tf_prefix + "/" + map_frame).c_str();
+
+        // std::string fromFrameRel = (tf_prefix + "/" + map_frame).c_str();
+        // std::string toFrameRel = (tf_prefix + "/" + base_frame).c_str();
+
+        RCLCPP_INFO(get_logger(),  "fromFrameRel topic (target_frame): %s", fromFrameRel.c_str());
+        RCLCPP_INFO(get_logger(),  "toFrameRel topic (source_frame): %s",toFrameRel.c_str());
+
+        geometry_msgs::msg::TransformStamped tf_transform;
+        bool tf_ok = true;
+
+        try {
+            tf_transform = tf_buffer_->lookupTransform(
+                toFrameRel, fromFrameRel,
+                tf2::TimePointZero);
+        } catch (const tf2::TransformException & ex) {
+            RCLCPP_INFO(
+                get_logger(), "Could not transform %s to %s: %s",
+                toFrameRel.c_str(), fromFrameRel.c_str(), ex.what());
+            tf_ok = false;
+            return;
+        }
 
-      pose_stamped.pose.pose.orientation.x = transform.getRotation().getX();
-      pose_stamped.pose.pose.orientation.y = transform.getRotation().getY();
-      pose_stamped.pose.pose.orientation.z = transform.getRotation().getZ();
-      pose_stamped.pose.pose.orientation.w = transform.getRotation().getW();
+        if(tf_ok)
+        {
+            RCLCPP_INFO(get_logger(),  "TF okey");
 
-      pose_publisher.publish(pose_stamped);
+            geometry_msgs::msg::PoseWithCovarianceStamped pose_stamped;
+            pose_stamped.header.stamp = rclcpp::Clock{}.now();
```

Comment:
should be used the original timepoint from ``lookupTransform` method. which is given `geometry_msgs::msg::TransformStamped`